reference_frame: "odom"
wheelRadius: 0.08
wheelSeperator: 0.48
mpc:
  motionTypes:
    - "/navigation"
    - "/navigation"
  goals:
    - [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, -1.0, 0.0, 2.0, 0.0]
    - [0.0, 2.0, 0.0, 0.0, 0.0, 0.0, -1.0, 0.0, 2.0, 0.0]
  timeStep: 0.5
  maxError: 0.1
  rate: 2.0
  noOrientation:
    safetyMargin: 0.05
    #arm, pos, orientation, slack, u, q_dot
    weights: [0.5, 10.0, 0.0, 10000000.0, 0.4, 10.0]
    #pos, orientation, arm
    errorWeights: [0.5, 0.0, 0.5]
  navigation:
    safetyMargin: 0.05
    #arm, pos, orientation, slack, u, q_dot
    weights: [0.5, 10.0, 0.5, 10000000.0, 0.4, 10.0]
    errorWeights: [0.5, 0.5, 0.5]
  fineMotion:
    safetyMargin: 0.01
    weights: [2.5, 50.0, 2.5, 10000000.0, 0.4, 10.0]
    errorWeights: [1.0, 1.0, 1.0]
  armMotion:
    # Negative safetyMargin to allow the gripper to get close to the table, sort of a hack
    safetyMargin: -0.8
    weights: [1.0, 0.0, 0.0, 10000000.0, 10000000.0, 5.0]
    errorWeights: [0.0, 0.0, 10.0]
